### ISSUE TITLE
Compiler server leaks

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#6235]
Recently I'm often getting ouf of memory when working on my project with compilation server.
Looking at `--display memory` I found the following:

```
h3d.prim.Polygon : 159 KB
      LEAK h3d.prim.Cube
      LEAK h3d.prim.Grid
```

This seems like a false positive, clearing cl_descendants for all classes before entering display memory should work good enough (that's what cl_restore does already)

```
hxbit.NetworkSerializable : 76.2 MB
      LEAK (???)
      LEAK haxe.macro.Compiler(b96399064b57aaf61d225f1b3ba9c033)
      LEAK haxe.macro.JSGenApi(b96399064b57aaf61d225f1b3ba9c033)
      ...
```

This one is more problematic, it seems that our classes are leaking their macro context for reasons that need to be understood and fixed.

**Comment by Simn** [comment]
Did the print context change fix this?

**Comment by ncannasse** [comment]
I have fixed the false positive reports in 5e71046df397048c3c6330b2cd7f560fe796a719, the memory is still leaking on our daily dev, I need to investigate the possible source but i'm not sure where to start.
